File: kotlin/Easy/PalindromeNumber.kt
package Easy

class PalindromeNumber{
    fun isPalindrome(x: Int): Boolean{
        if (x == 0) return true
        if (x < 0 || x%10 == 0) return false
        var a = x
        var b = 0
        while(a > b){
            b = b*10 + a%10
            a /= 10
        }
        return a == b || a == b/10
    }
}
fun main () {
    var a = PalindromeNumber().isPalindrome(1234321)
    println(a)
}

File: javascript/Easy/PalindromeNumber.js
/**
 * @param {number}
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x == 0) return true;
    if(x < 0 || x%10 == 0){
        return false;
    }
    let a = 0;
    while(x > a){
        a = a * 10 + x%10
        x = parseInt(x/10)
    }
    return (x == a || x == parseInt(a/10))
}

console.log(isPalindrome(12321))